types/defaultTo.d.ts

```diff
@@ -1,2 +1,12 @@
-export function defaultTo<T, U>(a: T, b: U | null | undefined): T | U;
-export function defaultTo<T>(a: T): <U>(b: U | null | undefined) => T | U;
+
+export function defaultTo<Default, Value>(a: Default, b: Value): (Value extends (null | undefined) ? Default | Exclude<Value, null | undefined> : Value);
```

Comment:
`Exclude<Value, null | undefined>` I believe is the same as `NonNullable<Value>`, might as well use the

Reply:
Yeah, I had `NonNullable<Value>` at one point but thought it was clearer to use `Exclude` because the `null | undefined` matches in the `extends`, whereas that connection is obscured using `NonNullable`.  I'm fine with either.